src/features/cart/cart.spec.ts

```diff
@@ -0,0 +1,143 @@
+import {addItem, removeItem, addDiscount, getCartItemPrice, getTotalPrice, getCartItems, cart, discountCodes} from "./cartModule"
+
+
+test("addItem increaments quantity given existing item", () => {
+
+    // Given 
+    const item1 = {
+            id: 1, 
+            title: "Item 1",
+            price: 5,
+            quantity: 1,
+            discountApplied : []
+        
+    }
+    cart.items.push(item1)
+
+    // when 
+    addItem(item1)
+
+    // effect
+
+    expect(cart.items[0].quantity).toEqual(2)
+})
+
+test("addItem adds item to cart given non-existing item", () => {
+    // Given 
+    const item = {
+        id: 2,
+        title: "item 2",
+        price: 10,
+        quantity: 5,
+        discountApplied: []
+    }
+
+    // when 
+    addItem(item)
+
+    //effect
+    expect(cart.items).toContain(item)
+
+})
+
+test("removeItem decreaments quantity given item with more than 1 quantity", () => {
+    //Given
+    const id = 1;
+
+    // when 
+
+    removeItem(id);
+
+    // effect
+    expect(cart.items[0].quantity).toBe(1)
```

Comment:
One test should not typically depend on another test case. Looks like this test case is relying on the addItem test case to have an item added, we should remove that dependency. 

So, best would be using `afterEach()` to clean up cart items.
https://jestjs.io/docs/setup-teardown

Make sure that a test case succeeds when it is the only test that runs.
https://jestjs.io/docs/setup-teardown#general-advice